pkg/cmd/server/api/validation/master.go

```diff
@@ -157,6 +157,15 @@ func ValidateMasterConfig(config *api.MasterConfig, fldPath *field.Path) Validat
 			}
 		}
 	}
+	if len(config.NetworkConfig.IngressIPNetworkCIDRs) > 1 {
+		validationResults.AddErrors(field.Invalid(fldPath.Child("networkConfig", "ingressIPNetworkCIDRs"), config.NetworkConfig.IngressIPNetworkCIDRs, "must be a single valid CIDR notation IP range (e.g. 172.30.0.0/16)"))
+	}
+	if len(config.NetworkConfig.IngressIPNetworkCIDRs) == 1 {
+		cidr := config.NetworkConfig.IngressIPNetworkCIDRs[0]
+		if _, _, err := net.ParseCIDR(cidr); err != nil {
+			validationResults.AddErrors(field.Invalid(fldPath.Child("networkConfig", "ingressIPNetworkCIDRs").Index(0), cidr, "must be a valid CIDR notation IP range (e.g. 172.30.0.0/16)"))
+		}
+	}
```

Comment:
Currently, we are checking for IP overlap between ClusterNetworkCIDR and ServiceNetworkCIDR inside the SDN plugin. Looks like this is the good place to ensure various cidrs are disjoint. Detecting these potential issues can avoid unnecessary debugging later. Could be done as a separate PR.